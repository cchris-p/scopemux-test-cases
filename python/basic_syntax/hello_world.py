#!/usr/bin/env python3
"""
A simple "Hello, World!" program in Python

This example demonstrates basic Python syntax including:
- Comments and docstrings
- Function definitions
- Print statements
- Conditional execution
"""


def main():
    """
    Program entry point
    """
    print("Hello, World!")


if __name__ == "__main__":
    # This block executes when the script is run directly
    main()
