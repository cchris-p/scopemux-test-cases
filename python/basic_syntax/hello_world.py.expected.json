{
  "language": "Python",
  "ast": {
    "type": "module",
    "functions": [
      {
        "name": "main",
        "type": "function_definition",
        "parameters": [],
        "body_range": {
          "start_line": 9,
          "start_column": 0,
          "end_line": 13,
          "end_column": 4
        },
        "docstring": "Program entry point"
      }
    ],
    "comments": [
      {
        "type": "line_comment",
        "text": "This block executes when the script is run directly",
        "line": 16
      }
    ],
    "docstrings": [
      {
        "type": "module_docstring",
        "text": "A simple \"Hello, World!\" program in Python\n\nThis example demonstrates basic Python syntax including:\n- Comments and docstrings\n- Function definitions\n- Print statements\n- Conditional execution",
        "range": {
          "start_line": 1,
          "start_column": 0,
          "end_line": 8,
          "end_column": 3
        }
      },
      {
        "type": "function_docstring",
        "text": "Program entry point",
        "function": "main",
        "range": {
          "start_line": 10,
          "start_column": 4,
          "end_line": 12,
          "end_column": 7
        }
      }
    ],
    "function_calls": [
      {
        "name": "print",
        "arguments": [
          "\"Hello, World!\""
        ],
        "line": 12
      },
      {
        "name": "main",
        "arguments": [],
        "line": 17
      }
    ],
    "conditionals": [
      {
        "type": "if_statement",
        "test": "__name__ == \"__main__\"",
        "line": 15
      }
    ]
  }
}
